### PR TITLE
Parameterized name as suffix for all resources tag

### LINKED ISSUE #29: [Blueprints, 01-getting-started] Tag Invalid for_each argument
On the first apply, I encountered this error:

```
│ Error: Invalid for_each argument
│
│   on .terraform/modules/eks/main.tf line 97, in resource “aws_ec2_tag” “cluster_primary_security_group”:
│   97:   for_each = { for k, v in merge(var.tags, var.cluster_tags) :
│   98:     k => v if local.create && k != “Name” && var.create_cluster_primary_security_group_tags && v != null
│   99:   }
│     ├────────────────
│     │ local.create is true
│     │ var.cluster_tags is empty map of string
│     │ var.create_cluster_primary_security_group_tags is true
│     │ var.tags is map of string with 2 elements
│
│ The “for_each” map includes keys derived from resource attributes that cannot be determined until apply, and so Terraform cannot determine the full set of keys that will
│ identify the instances of this resource.
│
│ When working with unknown values in for_each, it’s better to define the map keys statically in your configuration and place apply-time results only in the map values.
│
│ Alternatively, you could use the -target planning option to first apply only the resources that the for_each value depends on, and then apply a second time to fully converge.
```

I noticed that the tags key-value collections was taken from locals:
```
locals {
  name   = "cbci-bp01-i${random_integer.ramdom_id.result}"
  ...
  tags = merge(var.tags, {
    "tf:blueprint"  = local.name
    "tf:repository" = "github.com/cloudbees/terraform-aws-cloudbees-ci-eks-addon"
  })
}
```

During the first apply, the value of `random_integer.ramdom_id.result` is not yet known, which triggers the error above.

### PR DESCRIPTION
**Pull request by wellsiau-aws**
Resolve #29 

Since `for_each` cannot resolve value from random-id provider, the best way to address this is by letting customer to choose the random id. This will keep it deterministic for each plan / apply.

I also externalized other parameters such as EKS version and CIDR. This allows end-user to use the module as-is from abstraction such as AWS Service Catalog or Terraform Cloud no-code provisioning without making changes to the locals.

### CODE REVIEW

File: blueprints/01-getting-started/variables.tf [hunk 1]
```diff
@@ -18,3 +32,19 @@ variable "temp_license" {
   description = "Temporary license details"
   type        = map(string)
 }
+
+variable "k8s_version" {
```
[the remark this answers is not in the document]
Comment (wellsiau-aws) [reply blueprints/01-getting-started/variables.tf:36]: This variable makes it easier for user to run test without modifying the underlying .tf files. 

That said, I recognized the concern of externalizing other cluster / EKS level attributes, which are not the purpose of this reference blueprint.

File: blueprints/01-getting-started/variables.tf [hunk 2]
```diff
@@ -18,3 +32,19 @@ variable "temp_license" {
...
+  description = "EKS cluster version, refer to: https://docs.cloudbees.com/docs/cloudbees-common/latest/supported-platforms/cloudbees-ci-cloud#_kubernetes "
+  type        = string
+  default     = "1.26"
+}
+
+variable "vpc_cidr" {
```
[the remark this answers is not in the document]
Comment (wellsiau-aws) [reply blueprints/01-getting-started/variables.tf:42]: same as above

File: blueprints/01-getting-started/main.tf [hunk 3]
```diff
@@ -5,18 +5,18 @@ data "aws_route53_zone" "this" {
 data "aws_availability_zones" "available" {}
 
 locals {
-  name   = "cbci-bp01-i${random_integer.ramdom_id.result}"
-  region = "us-east-1"
+  name   = "cbci-bp-${var.name}"
```
[the remark this answers is not in the document]
Comment (wellsiau-aws) [reply blueprints/01-getting-started/main.tf:8]: sounds good, I will add the suggestion

**Comment by carlosrodlop** [comment]
It has been addressed in #36

**Pull request closed without merging**